etna/lib/commands.rb

```diff
@@ -114,38 +105,111 @@ class Create < Etna::Command
         include WithEtnaClients
         include WithLogger
 
-        def execute(filepath)
-          create_project_workflow = Etna::Clients::Magma::CreateProjectFromJsonWorkflow.new(
+        def execute(project_name, project_name_full)
+          create_args = {project_name: project_name, project_name_full: project_name_full}
+          Etna::Clients::Magma::ProjectValidator.new(**create_args).validate!("#{program_name} args invalid!")
+
+          create_project_workflow = Etna::Clients::Magma::CreateProjectWorkflow.new(
               magma_client: magma_client,
               janus_client: janus_client,
-              filepath: filepath)
+              **create_args
+          )
           create_project_workflow.create!
         end
       end
+    end
+
+    class Models
+      include Etna::CommandExecutor
 
-      class Validate < Etna::Command
+      class Add < Etna::Command
+        include WithEtnaClients
         include WithLogger
 
-        def execute(filepath)
-          user_json = JSON.parse(File.read(filepath))
-          magma_json = Etna::Clients::Magma::ProjectConverter.convert_project_user_json_to_magma_json(user_json)
-          project = Etna::Clients::Magma::Project.new(magma_json)
+        boolean_flags << '--execute'
+        string_flags << '--file'
+        string_flags << '--target-model'
 
-          validator = Etna::Clients::Magma::ProjectValidator.new(project)
-          validator.validate
+        def execute(project_name, execute: false, target_model: 'project', file: "#{project_name}_models_#{target_model}_tree.csv")
+          reset
 
-          return puts "Project JSON is well-formatted!" if validator.valid?
+          unless File.exists?(file)
+            puts "File #{file} is being prepared from the #{project_name} project."
+            puts "Copying models descending from #{target_model}..."
+            prepare_template(file, project_name, target_model)
+            puts
+            puts "Done!  You can start editing the file #{file} now, and I will report validation errors here."
+          end
 
-          puts "Project JSON has #{validator.errors.length} errors:"
-          validator.errors.each do |error|
-            puts "  * " + error.gsub("\n", "\n\t")
+          load_models_from_csv(file)
+
+          while true
```

Comment:
Polling watch loop that validates the file and allows executing the file with confirmation once it passes validation.